ValkeyLDAP.md

```diff
@@ -0,0 +1,271 @@
+
+- LDAP connection objects with state tracking
+- Priority queue based on connection health
+- Automatic connection cycling
+- Size limits and idle timeouts
+
+### Configuration
+
+#### Server Configuration
+
+- LDAP.servers: list of LDAP server addresses (comma-separated)
+- LDAP.port: LDAP port (default: 636 for LDAPS)
+- LDAP.use_ssl: whether to use LDAPS (default: true)
+- LDAP.timeout.connect: connection timeout in ms (default: 1000)
+- LDAP.timeout.search: search operation timeout in ms (default: 2000)
+- LDAP.retry.max: maximum retry attempts (default: 3)
+- LDAP.retry.delay: delay between retries in ms (default: 500)
+
+#### Authentication Configuration
+
+- LDAP.basedn: base DN for user searches
+- LDAP.binddn: service account DN for initial bind
+- LDAP.bindpw: service account password (stored securely)
+- LDAP.user_filter: LDAP filter for user searches
+- LDAP.group_filter: LDAP filter for group membership queries
+- LDAP.default_permissions: default ACL for authenticated users
+
+#### Cache Configuration
+
+- LDAP.cache.enabled: enable credential caching (default: true)
+- LDAP.cache.ttl: cache entry TTL in seconds (default: 300)
+- LDAP.cache.size: maximum cache size (default: 10000)
+- LDAP.cache.persist: persist cache across restarts (default: false)
+- LDAP.cache.encryption: encryption method for cached credentials
+
+#### Security Configuration
+
+- LDAP.tls.cert: path to client certificate
+- LDAP.tls.key: path to client key
+- LDAP.tls.ca: path to CA certificate
+- LDAP.tls.verify: verify server certificates (default: true)
+- LDAP.tls.ciphers: allowed cipher suites
+- LDAP.tls.version: minimum TLS version (default: TLS1.2)
+
+### Metrics and Monitoring
+
+#### Exported Metrics
+- Authentication attempts (success/failure)
+- Cache performance (hit ratio, size, evictions)
+- LDAP connection status (latency, errors, timeouts)
+- Group resolution performance
+- Resource utilization (memory, CPU)
+
+#### Health Indicators
+- LDAP server connectivity
+- Cache integrity
+- Configuration consistency
+- Error rates and patterns
+- Response time percentiles
+
+### Error Handling and Logging
+
+#### Error Categories
+- Configuration errors (invalid parameters, inconsistent settings)
+- Connection errors (network failures, timeouts, rejected connections)
+- Authentication errors (invalid credentials, expired accounts)
+- Authorization errors (insufficient permissions, group mapping failures)
+- Internal errors (memory allocation, thread management)
+
+#### Logging Levels
+
+- ERROR: critical issues requiring immediate attention
+- WARN: potential issues that don't prevent operation
+- INFO: normal operational events
+- DEBUG: detailed debugging information
+- TRACE: full protocol-level logging for troubleshooting
+
+### Cluster mode
+
+TBD :
+- this should follow the existing authentication rules for Valkey Cluster
+- configuration should be sync-ed across all nodes
+- authentication cache should be consistent across nodes 
```

Comment:
In AWS we don't have a consistent authentication cache and haven't seen major issues, but it wouldn't be hard to do that with the clusterbus.